tests/correctness.rs

```diff
@@ -23,24 +49,40 @@ fn correctness() {
         IterationControl::Break
     });
 
-    let ip = (test_function as u64).wrapping_sub(bias.unwrap());
+    let test = |sym: u64, expected_prefix: &str| {
+        let ip = sym.wrapping_sub(bias.unwrap());
+
+        let frames = ctx.find_frames(ip).unwrap();
+        let frame = frames.last().unwrap().unwrap();
+        let name = frame.function.as_ref().unwrap().demangle().unwrap();
+        // Old rust versions generate DWARF with wrong linkage name,
+        // so only check the start.
+        if !name.starts_with(expected_prefix) {
+            panic!("incorrect name '{}', expected {:?}", name, expected_prefix);
+        }
+    };
+
+    test(test_function as u64, "correctness::test_function");
+    test(
+        small::test_function as u64,
+        "correctness::small::test_function",
+    );
+    test(aux::foo as u64, "aux::foo");
```

Comment:
What are these two tests checking that wasn't covered before?

Reply:
The `small` module was me attempting to reproduce the original issue and the `aux::foo` test will fail on macOS before this PR but succeeds after this PR. I'm not entirely sure what makes rustc generate the object that causes this but w/e it is it seems to work.